### PR TITLE
New package: EternalTerminal-5.1.9

### PR DESCRIPTION
**Pull request by ndowens**
Reconnects on SSH disconnects

### CODE REVIEW

**Comment by ndowens** [comment]
Got to add LICENSE and sv file

File: srcpkgs/EternalTerminal/template [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'EternalTerminal'
+pkgname=EternalTerminal
+version=5.1.8
+revision=1
+wrksrc="${pkgname}-et-v${version}"
+build_style=cmake
+hostmakedepends=""
+makedepends="gflags-devel libsodium-devel protobuf-devel"
+depends=""
```
Comment (abenson) [inline srcpkgs/EternalTerminal/template]: Remove this

File: srcpkgs/EternalTerminal/template [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'EternalTerminal'
+pkgname=EternalTerminal
+version=5.1.8
+revision=1
+wrksrc="${pkgname}-et-v${version}"
+build_style=cmake
+hostmakedepends=""
```
Comment (abenson) [inline srcpkgs/EternalTerminal/template]: Remove this

Comment (ndowens) [reply srcpkgs/EternalTerminal/template]: Not done working on it yet

**Comment by abenson** [comment]
Shouldn't need to add LICENSE for Apache-2.0

**Comment by Johnnynator** [comment]
Smalle changes that would get it working on musl
```
diff --git a/srcpkgs/EternalTerminal/patches/musl.patch b/srcpkgs/EternalTerminal/patches/musl.patch
new file mode 100644
index 0000000000..3c8bd66cd7
--- /dev/null
+++ b/srcpkgs/EternalTerminal/patches/musl.patch
@@ -0,0 +1,21 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index 3945f33..dd5b572 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -3,6 +3,7 @@ project (EternalTCP VERSION 5.1.8)
+ 
+ option(BUILD_TEST "Build all unit test" OFF)
+ option(CODE_COVERAGE "Enable code coverage" OFF)
++option(EXECINFO "Enable use of libexecinfo" OFF)
+ 
+ SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DET_VERSION='\"${PROJECT_VERSION}\"'")
+ # For easylogging, disable default log file, enable crash log, ensure thread safe, and catch c++ exceptions
+@@ -88,6 +89,8 @@ IF(FREEBSD)
+   set(CORE_LIBRARIES util execinfo)
+ ELSEIF(NETBSD)
+   set(CORE_LIBRARIES util resolv execinfo)
++ELSEIF(EXECINFO)
++  set(CORE_LIBRARIES util resolv execinfo)
+ ELSE()
+   set(CORE_LIBRARIES util resolv)
+ ENDIF()
diff --git a/srcpkgs/EternalTerminal/template b/srcpkgs/EternalTerminal/template
index 1147367540..c607021abd 100644
--- a/srcpkgs/EternalTerminal/template
+++ b/srcpkgs/EternalTerminal/template
@@ -15,3 +15,8 @@ distfiles="https://github.com/MisterTea/EternalTerminal/archive/et-v${version}.t
 checksum=da30ae120fd5bda3939909c27a91a857f23cfe0648b7e3905dfb0b89f91a82e8
 
 LDFLAGS="-lgflags"
+
+case $XBPS_TARGET_MACHINE in
+	*-musl) configure_args="-DEXECINFO=TRUE"
+		makedepends+=" libexecinfo-devel";;
+esac
```

**Comment by ndowens** [comment]
Awesome, will try the musl patch soon

**Comment by Johnnynator** [comment]
For Cross Compiling adding `protobuf` to hostmakedepends should be enough

File: srcpkgs/EternalTerminal/template [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+LDFLAGS="-lgflags"
+
+# patch manually, as libexecinfo is musl only
+case $XBPS_TARGET_MACHINE in
+	*-musl)
+	configure_args="-DEXECINFO=TRUE"
+	makedepends+=" libexecinfo-devel"
+	post_extract() {
+		patch -Np0 < ${FILESDIR}/musl.patch
```
[the remark this answers is not in the document]
Comment (ndowens) [reply srcpkgs/EternalTerminal/template]: It gave some execinfo error if I tried installing it though xi.

File: srcpkgs/EternalTerminal/template [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+makedepends="gflags-devel libsodium-devel protobuf-devel"
+depends="gflags-devel libsodium-devel protobuf-devel"
+short_desc="Re-Connectable secure remote shell"
+maintainer="Nathan Owens <ndowens04@gmail.com>"
+license="Apache-2.0"
+homepage="https://mistertea.github.io/EternalTerminal"
+distfiles="https://github.com/MisterTea/EternalTerminal/archive/et-v${version}.tar.gz"
+checksum=da30ae120fd5bda3939909c27a91a857f23cfe0648b7e3905dfb0b89f91a82e8
+
+system_accounts="eternal"
```
Comment (maxice8) [inline srcpkgs/EternalTerminal/template]: please use account prefixed with underscore

Comment (ndowens) [reply srcpkgs/EternalTerminal/template]: Done

**Comment by maxice8** [comment]
EternalWIP ?

**Comment by ndowens** [comment]
Updated

**Comment by Duncaen** [comment]
Not a fan of the conditional musl patch.

**Comment by maxice8** [comment]
Pretty much what @Duncaen said

**Comment by ndowens** [comment]
Done, for some reason I didn't get the error I did before, the reason for the conditional patch.

**Comment by ndowens** [comment]
https://travis-ci.org/ndowens/void-packages/builds/482891082 Seems to build fine

**Pull request merged**